keps/0001-single-controller-multi-proxy.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Single Controller Multi Proxy DBLESS
+status: provisional
+---
+
+# Decoupled KIC Components
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Historically the [Kong Kubernetes Ingress Controller (KIC)][kic] has [required
+a controller pod to be present per proxy pod in DBLESS mode][kic702] and
+generally had all components coupled tightly together in a single `Pod`. This
+KEP aims to make it possible to deploy a single controller to manage any number
+of proxy instances in DBLESS mode.
+
+[kic]:https://github.com/kong/kubernetes-ingress-controller
+[kic702]:https://github.com/Kong/kubernetes-ingress-controller/issues/702
+
+## Motivation
+
+- reduce Kubernetes API overhead in high scale Kong environments where large
+  numbers of proxies are needed in DBLESS mode
+- better support an operator use case based on the `Gateway` API to enable a
+  single KIC to manage multiple `Gateways`.
+
+[gateway-api]:https://kubernetes-sigs.github.io/gateway-api/
+
+### Goals
+
+- enable TLS and authentication for DBLESS `proxy` pods
+- enable multiple `proxy` instances to be managed by a single KIC
+- provide service discovery and automatic configuration for new `proxy` pods
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an operator with DBLESS deployed Kong I want a single controller to be able
+to manage my proxy instances.
```

Comment:
I think that it makes sense for user stories to include a justification, like the usual format "as Y I want Y **so that Z**" suggests. This can help us express the user expectations on a more problem-oriented basis (as opposed to solution-oriented).

Examples on my mind:
- As a Kubernetes cluster operator I want a single controller to write config to multiple kong gateways, so that scaling out my gateway setup doesn't linearly increase the Kubernetes Admin API load.
- As a Kubernetes cluster operator I want the Admin API traffic between KIC and Gateway to be encrypted, so that I can safely allow KIC to talk to the Admin API between pods.

Reply:
Going with a combination of the two suggestions:

```suggestion
As an operator with DBLESS deployed Kong I want a single controller to be able
to manage my proxy instances so that scaling out my gateway setup doesn't
linearly increase the Kubernetes Admin API load and I can safely allow KIC to talk
 to the Admin API between pods.
```